crypto/evp_extra/p_pqdsa.c

```diff
@@ -56,9 +56,9 @@ static int pkey_pqdsa_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY *pkey) {
   return 1;
 }
 
-static int pkey_pqdsa_sign_message(EVP_PKEY_CTX *ctx, uint8_t *sig,
-                                     size_t *sig_len, const uint8_t *message,
-                                     size_t message_len) {
+static int pkey_pqdsa_sign_generic(EVP_PKEY_CTX *ctx, uint8_t *sig,
+                                   size_t *sig_len, const uint8_t *message,
+                                   size_t message_len, int prehash) {
```

Comment:
To make it clear that this is not HashML-DSA, I suggest replacing `prehash` with `external_mu`.

Reply:
Not a good idea. This is the `PQDSA` logic for ALL PQDSA types. ExternalMu isn't defined for any PQDSA scheme other than ML-DSA. For example, why would SLH-DSA have a flag for ExternalMu? 

I have changed all the ML-DSA specific uses of a pre-hash flag to use `external_mu` but the ones that remain called `prehash` are intentional so that we can use the flag for the purposes of indicating any pre-hash method for any PQDSA signature scheme.

Reply:
I agree this shouldn't be external_mu, but it isn't clear if prehash means the input has been pre-hashed or if AWS-LC should hash the input before signing it, can this follow the method names and use sign_message or sign_digest 0/1?

Reply:
Thanks Andrew, have I understood you correctly with this update: [318f0c1](https://github.com/aws/aws-lc/pull/2113/commits/318f0c1c0835cbcd3c30643cb25bdc567c566926)